File: Python Excel Examples/WorksheetsApiDemo/protect/protectWorksheet.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.worksheets_api.WorksheetsApi(configuration)

name = "ProtectWorksheet.xlsx"
storage = ""
folder = "ExcelDocument"
sheetName = "Sheet1"
protectParameter = spirecloudexcel.models.ProtectSheetParameter()
protectParameter.password = "123"
protectParameter.protection_type = "scenarios"
protectParameter.allow_deleting_column = "true"
protectParameter.allow_selecting_locked_cell = "true"
protectParameter.allow_formatting_row = "true"
api.protect_worksheet(name, sheetName, protectParameter, storage=storage, folder=folder)

File: JavaScript Excel Examples/WorksheetsApiDemo/Protect/protectWorksheet.js
(function (){
    var appId = "your id";
    var appKey = "your key"; 
    
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudExcel = require('../../../src/index');
    var apiClient = new SpirecloudExcel.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudExcel.WorksheetsApi(apiClient);

    var name = "ProtectWorksheet.xlsx";
    var sheetName = "Sheet3";
    var protectParameter = new SpirecloudExcel.ProtectSheetParameter();
    protectParameter.password = 123;
    protectParameter.protectionType = "scenarios";
    protectParameter.allowSelectingLockedCell = true;
    protectParameter.allowDeletingColumn = true;
    protectParameter.allowFormattingRow = true;
    var opts = {
        folder: "input"
    };
    instance.protectWorksheet(name, sheetName, protectParameter, opts, function (error, data, response) {
            if (error) throw error;
        });
})();